crates/bevy_ui/src/render/mod.rs

```diff
@@ -984,6 +987,125 @@ pub fn extract_text_shadows(
     }
 }
 
+fn extract_text_outlines(
+    mut commands: Commands,
+    mut extracted_uinodes: ResMut<ExtractedUiNodes>,
+    texture_atlases: Extract<Res<Assets<TextureAtlasLayout>>>,
+    uinode_query: Extract<
+        Query<(
+            Entity,
+            &ComputedNode,
+            &ComputedNodeTarget,
+            &UiGlobalTransform,
+            &InheritedVisibility,
+            Option<&CalculatedClip>,
+            &TextLayoutInfo,
+            &TextOutline,
+        )>,
+    >,
+    camera_map: Extract<UiCameraMap>,
+) {
+    let mut start = extracted_uinodes.glyphs.len();
+    let mut len = 0;
+
+    let mut camera_mapper = camera_map.get_mapper();
+    for (
+        entity,
+        uinode,
+        target,
+        global_transform,
+        inherited_visibility,
+        clip,
+        text_layout_info,
+        outline,
+    ) in &uinode_query
+    {
+        // Skip if not visible or if size is set to zero (e.g. when a parent is set to `Display::None`)
+        if !inherited_visibility.get() || uinode.is_empty() {
+            continue;
+        }
+
+        let Some(extracted_camera_entity) = camera_mapper.map(target) else {
+            continue;
+        };
+
+        let width = outline.width as i32;
+        let aa_factor = outline.anti_aliasing.unwrap_or(1.0);
+        let color: LinearRgba = outline.color.into();
+
+        for (
+            i,
+            PositionedGlyph {
+                position,
+                atlas_info,
+                ..
+            },
+        ) in text_layout_info.glyphs.iter().enumerate()
+        {
+            let rect = texture_atlases
+                .get(&atlas_info.texture_atlas)
+                .unwrap()
+                .textures[atlas_info.location.glyph_index]
+                .as_rect();
+
+            for offset_x in -width..=width {
+                for offset_y in -width..=width {
+                    if offset_x == 0 && offset_y == 0 {
+                        continue;
+                    }
+
```

Comment:
```suggestion
                    if width * width < offset_x * offset_x + offset_y * offset_y {
                        continue;
                    }
```

This uses 21% less glyphs and the outlines mostly look a lot better:

before:
![sharp](https://github.com/user-attachments/assets/6925bdcb-aa5b-4d52-8f8d-3039182467a6)

after:
![rounded](https://github.com/user-attachments/assets/6d4c207b-2128-45e4-82bb-a4992f677acb)